app/core/steps/AddressStep.js

```diff
@@ -16,15 +16,22 @@ class AddressStep extends ValidationStep {
         return [ctx];
     }
 
-    handlePost(ctx, errors) {
+    handlePost(ctx, errors, formdata, session) {
         ctx.address = ctx.postcodeAddress || ctx.freeTextAddress;
         ctx.postcode = ctx.postcode ? ctx.postcode.toUpperCase() : ctx.postcode;
         if (!ctx.postcodeAddress) {
             delete ctx.addresses;
         }
+        session.addresses = session.addresses || {};
```

Comment:
Can we have a unit test to test that if `session.addresses` doesn't exist then it gets set to `{}`?

Reply:
yes, i think is covered in testAddressStep.js in "it('when freeTextAddress exists',(done..." where it is checked there is a session.addresses{} after the handle post.

Reply:
👍